File: ProgramaTiendaPython/Compra.py
from datetime import datetime

class Compra:
    def __init__(self, cliente, productos, cantidad, total, fecha_compra=None):
        self.cliente = cliente
        self.productos = productos
        self.cantidad = cantidad
        self.total = total
        self.fecha_compra = fecha_compra if fecha_compra else datetime.now().strftime("%Y-%m-%d")

    def agregar_producto(self, producto):
        self.productos.append(producto)

    def getCliente(self):
        return self.cliente

    def obtener_informacion(self):
        return f"Compra realizada por el cliente {self.cliente.id}, productos comprados: {len(self.productos)}, total de la compra: {self.total}"

File: ProgramaTiendaJava/Compra.java
import java.util.ArrayList;

public class Compra {
    private Cliente cliente;
    private ArrayList<Producto> productos;
    private String fechaCompra;
    private String Producto;
    private int cantidad;
    private double total;

    // Constructor que incluye el cliente y los productos
    public Compra(Cliente cliente, ArrayList<Producto> productos, int cantidad, double total) {
        this.cliente = cliente;
        this.productos = productos;
        this.fechaCompra = java.time.LocalDate.now().toString();
        this.cantidad = cantidad;
        this.total = total;
    }

    // Métodos getter para el cliente y los productos
    public Cliente getCliente() {
        return cliente;
    }

    public ArrayList<Producto> getProductos() {
        return productos;
    }

    public int getCantidad() {
        return cantidad;
    }

    public double getTotal() {
        return total;
    }

    public String getProducto() {
        return Producto;
    }



    // Método para agregar un producto a la lista de productos de la compra
    public void agregarProducto(Producto producto) {
        productos.add(producto);
    }

    public String obtenerInformacion() {
        StringBuilder info = new StringBuilder();
        info.append("Cliente: ").append(cliente.getId()).append(", Productos:");
        for (Producto producto : productos) {
            info.append(" ").append(producto.getNombre());
        }
        info.append(", Fecha de compra: ").append(fechaCompra);
        return info.toString();
    }
}


